spec/services/email_output_service_spec.rb

```diff
@@ -0,0 +1,81 @@
+require_relative '../../app/services/email_output_service'
+require_relative '../../app/services/email_service'
+require_relative '../../app/value_objects/attachment'
+
+describe EmailOutputService do
+  subject(:service) { described_class.new(email_service: email_service_mock) }
+
+  let(:email_service_mock) { class_double(EmailService) }
+
+  let(:email_action) do
+    {
+      recipientType: 'team',
+      type: 'email',
+      from: 'form-builder@digital.justice.gov.uk',
+      to: 'bob.admin@digital.justice.gov.uk',
+      subject: 'Complain about a court or tribunal submission',
+      email_body: 'Please find an application attached',
+      include_pdf: include_pdf,
+      include_attachments: include_attachments
+    }
+  end
+
+  let(:include_pdf) { false }
+  let(:include_attachments) { false }
+
+  let(:attachments) do
+    [
+      Attachment.new(type: 'output', url: 'example.com/foo', mimetype: 'application/pdf', filename: 'form1', path: nil),
+      Attachment.new(type: 'output', url: 'example.com/bar', mimetype: 'application/json', filename: 'form2', path: nil)
+    ]
+  end
+
+  let(:pdf_attachment) do
+    Attachment.new(type: 'output', url: nil, mimetype: 'application/pdf', filename: 'a generated pff', path: nil)
+  end
+
+  before do
+    allow(email_service_mock).to receive(:send_mail)
+
+    service.execute(action: email_action, attachments: attachments, pdf_attachment: pdf_attachment, submission_id: 'an-id-2323')
+  end
+
+  it 'execute sends an email' do
+    expect(email_service_mock).to have_received(:send_mail).with(to: 'bob.admin@digital.justice.gov.uk',
+                                                                 from: 'form-builder@digital.justice.gov.uk',
+                                                                 subject: 'Complain about a court or tribunal submission {an-id-2323} [1/1]',
+                                                                 body_parts: { 'text/plain': 'Please find an application attached' },
+                                                                 attachments: []).once
+  end
+
+  context 'when a include_attachments is true' do
```

Comment:
Better to write these so they are more understandable by someone new to the system. Something like `when the submission requires attachments`.